runtime/src/main/scala/akka/grpc/internal/AkkaNettyGrpcClientGraphStage.scala

```diff
@@ -102,6 +102,10 @@ private final class AkkaNettyGrpcClientGraphStage[I, O](
         override def onMessage(message: O): Unit =
           callback.invoke(message)
         override def onClose(status: Status, trailers: Metadata): Unit = {
+          if (!matVal.isCompleted) {
+            // Trailers only response, first invoke onHeaders to setup the materialized value
+            onHeaders(trailers)
```

Comment:
So, `trailers` in this case is the response header, and it contains the trailers. There's no way to determine which headers are for which - of course, `grpc-status` and `grpc-message` etc are obviously trailers, but there could presumably be other things that we can't predict should be in one or the other. So, code that's expecting to read something from the response header needs this metadata made available as a response header, and code that's expecting to read something from the trailer needs the same metadata made available as a trailer. So, that's why I've passed it to both.